model/annotations/task_annotations_test.go

```diff
@@ -38,8 +38,118 @@ func TestGetLatestExecutions(t *testing.T) {
+	issue1 := IssueLink{URL: "https://issuelink.com", IssueKey: "EVG-1234", Source: &Source{Author: "annie.black"}}
+	issue2 := IssueLink{URL: "https://issuelink.com", IssueKey: "EVG-1234", Source: &Source{Author: "not.annie.black"}}
+	assert.NoError(t, db.Clear(Collection))
+	a := TaskAnnotation{TaskId: "t1", Issues: []IssueLink{issue1, issue2}}
+	assert.NoError(t, a.Insert())
+
+	assert.NoError(t, RemoveIssueFromAnnotation("t1", 0, issue1))
+	annotationFromDB, err := FindOneByTaskIdAndExecution("t1", 0)
+	assert.NoError(t, err)
+	assert.NotNil(t, annotationFromDB)
+	assert.Len(t, annotationFromDB.Issues, 1)
+	assert.Equal(t, "not.annie.black", annotationFromDB.Issues[0].Source.Author)
+}
+
+func TestAddSuspectedIssueToAnnotation(t *testing.T) {
+	assert.NoError(t, db.Clear(Collection))
+	issue := IssueLink{URL: "https://issuelink.com", IssueKey: "EVG-1234"}
+	assert.NoError(t, AddSuspectedIssueToAnnotation("t1", 0, issue, "annie.black"))
+
+	annotation, err := FindOneByTaskIdAndExecution("t1", 0)
+	assert.NoError(t, err)
+	assert.NotNil(t, annotation)
+	assert.Len(t, annotation.SuspectedIssues, 1)
+	assert.NotNil(t, annotation.SuspectedIssues[0].Source)
+	assert.Equal(t, UIRequester, annotation.SuspectedIssues[0].Source.Requester)
+	assert.Equal(t, "annie.black", annotation.SuspectedIssues[0].Source.Author)
+
+	assert.NoError(t, AddSuspectedIssueToAnnotation("t1", 0, issue, "not.annie.black"))
+	annotation, err = FindOneByTaskIdAndExecution("t1", 0)
+	assert.NoError(t, err)
+	assert.NotNil(t, annotation)
+	assert.Len(t, annotation.SuspectedIssues, 2)
+	assert.NotNil(t, annotation.SuspectedIssues[1].Source)
+	assert.Equal(t, "not.annie.black", annotation.SuspectedIssues[1].Source.Author)
+}
+
+func TestRemoveSuspectedIssueFromAnnotation(t *testing.T) {
+	issue1 := IssueLink{URL: "https://issuelink.com", IssueKey: "EVG-1234", Source: &Source{Author: "annie.black"}}
+	issue2 := IssueLink{URL: "https://issuelink.com", IssueKey: "EVG-1234", Source: &Source{Author: "not.annie.black"}}
+	assert.NoError(t, db.Clear(Collection))
+	a := TaskAnnotation{TaskId: "t1", SuspectedIssues: []IssueLink{issue1, issue2}}
+	assert.NoError(t, a.Insert())
+
+	assert.NoError(t, RemoveSuspectedIssueFromAnnotation("t1", 0, issue1))
+	annotationFromDB, err := FindOneByTaskIdAndExecution("t1", 0)
+	assert.NoError(t, err)
+	assert.NotNil(t, annotationFromDB)
+	assert.Len(t, annotationFromDB.SuspectedIssues, 1)
+	assert.Equal(t, "not.annie.black", annotationFromDB.SuspectedIssues[0].Source.Author)
+}
+
+func TestMoveIssueToSuspectedIssue(t *testing.T) {
+	issue1 := IssueLink{URL: "https://issuelink.com", IssueKey: "EVG-1234", Source: &Source{Author: "this will be overridden"}}
+	issue2 := IssueLink{URL: "https://issuelink.com", IssueKey: "EVG-1234", Source: &Source{Author: "evergreen user"}}
+	assert.NoError(t, db.Clear(Collection))
+	a := TaskAnnotation{TaskId: "t1", Issues: []IssueLink{issue1, issue2}, SuspectedIssues: []IssueLink{issue2}}
```

Comment:
We shouldn't ever have duplicates between these two fields, so having issue2 in both lists is probably not the best setup.

Reply:
Just laziness haha but I respect that